test/lit_tests/raising/llvm_to_affine_access.mlir

```diff
@@ -0,0 +1,66 @@
+// RUN: enzymexlamlir-opt %s --pass-pipeline="builtin.module(llvm-to-affine-access)" | FileCheck %s
+
+func.func @test_load_store_conversion(%arg0: !llvm.ptr<1>, %idx: i64) {
+  %0 = llvm.getelementptr inbounds %arg0[%idx] : (!llvm.ptr<1>, i64) -> !llvm.ptr<1>, i64
+  %1 = llvm.load %0 {alignment = 1 : i64} : !llvm.ptr<1> -> i64
+  %2 = llvm.mul %1, %1 : i64
+
+  llvm.store %2, %0 {alignment = 1 : i64} : i64, !llvm.ptr<1>
+
+  return
+}
+
+// CHECK-LABEL: func @test_load_store_conversion
+// CHECK-SAME: %[[ARG0:.*]]: !llvm.ptr<1>
+// CHECK-SAME: %[[ARG1:.*]]: i64
+// CHECK: %[[MEMREF:.*]] = "enzymexla.ataddr"(%[[ARG0]]) {{.*}} memref<?xi8, 1>
```

Comment:
The original load is typed as i64, can we preserve that here instead of doing an vector of i8?

Reply:
My idea to solve this issue was to have custom load op which does 
affine.vector_load_ty %memref[%idx] vector<4xi8> : memref<i8x?>, i64 -> f64

which basically combines the affine.vector_load and the bitcast afterwards in one operation.

Because the current vector load + bitcast approach may lose the information about the original type alignment etc and break things.